### PR TITLE
Chapter 9

### PR DESCRIPTION
**Pull request by thaidoandat**
image
link questions: https://docs.google.com/document/d/1GFPo8Xc29CrcNkbCtUSq69EirjnvgB_knkBdE4a0pgE/edit

### CODE REVIEW

**Comment by thaidoandat** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -3,19 +3,22 @@ def new; end
 
   def create
     user = User.find_by email: params[:session][:email].downcase
+    check = user&.authenticate params[:session][:password]
```
Comment (ngocvt-0484) [inline app/controllers/sessions_controller.rb]: valid_password

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class User < ApplicationRecord
+  attr_accessor :remember_token
```
Comment (ngocvt-0484) [inline app/models/user.rb:2]: attr_accessor để làm gì

**Comment by thaidoandat** [comment]
ready

File: app/models/user.rb [hunk 3]
```diff
@@ -18,6 +20,36 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  class << self
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+               BCrypt::Engine::MIN_COST
+             else
+               BCrypt::Engine.cost
+             end
```
Comment (ghost) [inline app/models/user.rb]:
```suggestion
      min_cost = ActiveModel::SecurePassword.min_cost
      cost = min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
```

**Review by ghost — COMMENTED** [review]
cmt

**Comment by thaidoandat** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**